passkeys-backend/assets/index.html

```diff
@@ -0,0 +1,336 @@
+            }
+            return buffer;
+        }
+
+        function stringToArrayBuffer(str) {
+            const encoder = new TextEncoder();
+            const uint8Array = encoder.encode(str);
+            return uint8Array.buffer;
+        }
+
+        const login = () => {
+            const authenticationCard = document.getElementById("container");
+            const passkeyCard = document.getElementById("modal");
+            authenticationCard.classList.add("invisible");
+            passkeyCard.classList.remove("invisible");
+        }
+
+        const signIn = async () => {
+            try {
+                const response = await fetch(`./authentication/start`);
+                const responseJSON = await response.json();
+
+
+                window.webauthnJSON.get(responseJSON)
+                    .then(async (publicKeyCredential) => {
+                        const { id, rawId, response, type, authenticatorAttachment } = publicKeyCredential
+                        const { authenticatorData, clientDataJSON, signature, userHandle } = response
+
+                        const authentication = await fetch('./authentication/verification', {
+                            method: "POST",
+                            headers: {
+                                "Content-Type": "application/json"
+                            },
+                            body: JSON.stringify({
+                                id: id,
+                                rawId: rawId,
+                                clientDataJson: clientDataJSON,
+                                authenticatorData: authenticatorData,
+                                signature: signature,
+                                userHandle: userHandle
+                            })
+                        });
+
+                        const authenticationJSON = await authentication.json();
+                        const {status, identity} = authenticationJSON
+                        if(status === "approved") {
+                            sessionStorage.setItem('session', identity);
+                            loadApp(identity);
+                        } else {
+                            console.log(status);
+                        }
+                    })
+                    .catch((err) => {
+                        if(err) {
+                            console.error("Something goes wrong or maybe you dont have a passkey for this application yet");
+                        }
+                    });
+            } catch (error) {
+                console.log(err);
+            }
+        }
+ 
+        const signUp = async () => {
+            const username = usernameElement.value
+
+            try {
+                const response = await fetch(`./registration/start`, {
+                    method: "POST",
+                    headers: {
+                        "Content-Type": "application/json"
+                    },
+                    body: JSON.stringify({
```

Comment:
nit but this can be JSON.stringify({ username })

Reply:
Done!